sappho/tilemap.py

```diff
@@ -67,28 +67,26 @@ def parse_rules(path_to_rules_file):
                 that are set
 
         """
-        
-        with open(path_to_rules_file) as f:
-            rules = f.readlines()
-
         tile_rules = {}
 
-        for rule in rules:
-            tile_ids_affected, flags = rule.split('=')
-            tile_ids_affected = [id_ for id_ in tile_ids_affected.split(',')]
-            flags = [flag.strip() for flag in flags.split(',')]
+        with open(path_to_rules_file) as f:
```

Comment:
I can't seem to find where it says that readlines is an iterator object? It seems to just to generate a list. Everything I've read says "automatically parses the read contents into a list of lines."

I feel like that's a needless optimization and it's premature.

I wanna focus on legibility over speediness. I avoid indentation when possible.

Optimize once there's a problem with bottleneck.